File: ios/calorietracker/Views/Coach/PromptGalleryText.swift
import Foundation

/// The gallery's text, keyed by catalog id (docs/coach.md §9).
///
/// The ids and the English source live in `shared/coach/prompt_gallery.json`; this table only
/// carries each one into the string catalog, so a card, a chip and a translation can never drift
/// apart. `PromptGalleryTextTests` compares this table with the catalog.
enum PromptGalleryText {
    static func title(_ id: String) -> String? {
        switch id {
        case "diet_chart_from_health": return String(localized: "Make a diet chart from my health data")
        case "macro_balance": return String(localized: "Are my macros balanced?")
        case "dinner_tonight": return String(localized: "What should I eat for dinner?")
        case "protein_sources": return String(localized: "Help me hit my protein target")
        case "week_review": return String(localized: "Review my week of eating")
        case "improve_sleep_cycle": return String(localized: "How do I improve my sleep cycle?")
        case "sleep_week": return String(localized: "How did I sleep this week?")
        case "sleep_consistency": return String(localized: "Is my sleep schedule consistent?")
        case "sleep_and_resting_hr": return String(localized: "Sleep and my resting heart rate")
        case "training_review": return String(localized: "Analyze my last 4 weeks of training")
        case "moving_enough": return String(localized: "Am I moving enough?")
        case "lift_progress": return String(localized: "How are my lifts progressing?")
        case "recovery_check": return String(localized: "Am I recovering between sessions?")
        case "explain_latest_report": return String(localized: "Explain my latest blood report")
        case "find_abnormal": return String(localized: "Find abnormal values in my reports")
        case "compare_reports": return String(localized: "Compare my last two reports")
        case "analyte_trend": return String(localized: "Chart my lab values over time")
        case "questions_for_doctor": return String(localized: "What should I ask my doctor?")
        case "adherence_check": return String(localized: "Am I taking my medicines on time?")
        case "medication_schedule": return String(localized: "Explain my current schedule")
        case "missed_dose_pattern": return String(localized: "Which doses do I keep missing?")
        case "reach_my_goal": return String(localized: "When will I reach my goal?")
        case "month_over_month": return String(localized: "Compare this month with last")
        case "one_thing_to_change": return String(localized: "One thing to change this month")
        case "whole_picture": return String(localized: "Give me the whole picture")
        default: return nil
        }
    }

    static func prompt(_ id: String) -> String? {
        switch id {
        case "diet_chart_from_health": return String(localized: "Make me a diet chart using my health data. Use what you know about my intake, weight trend and activity, and explain why each meal is there.")
        case "macro_balance": return String(localized: "How balanced are my macros over the last two weeks? Chart protein, carbs and fat and tell me what to change.")
        case "dinner_tonight": return String(localized: "Based on what I have already eaten today, what should I have for dinner to stay on target?")
        case "protein_sources": return String(localized: "I want more protein without many more calories. Look at what I usually eat and suggest realistic swaps.")
        case "week_review": return String(localized: "Review my last week of eating. What went well, what slipped, and what is the one thing to fix next week?")
        case "improve_sleep_cycle": return String(localized: "How do I improve my sleep cycle? Look at my bedtimes, wake times and sleep stages, chart the pattern, and give me a plan I can actually keep.")
        case "sleep_week": return String(localized: "How did I sleep this week? Chart it night by night and tell me whether it is enough.")
        case "sleep_consistency": return String(localized: "Is my sleep schedule consistent? Show me how much my bedtime moves around and what that is costing me.")
        case "sleep_and_resting_hr": return String(localized: "Does my resting heart rate change with how I sleep? Compare the two over the last month.")
        case "training_review": return String(localized: "Analyze my last four weeks of training. Where am I progressing, where have I stalled, and what should next week look like?")
        case "moving_enough": return String(localized: "Am I moving enough? Chart my steps and active energy for the last month against what would be a good target for me.")
        case "lift_progress": return String(localized: "Chart how my main lifts have moved over the last three months and tell me what to push next.")
        case "recovery_check": return String(localized: "Am I recovering well between sessions? Look at my training load alongside my sleep and resting heart rate.")
        case "explain_latest_report": return String(localized: "Explain my latest blood report: what stands out, what is normal, and what should I ask my doctor?")
        case "find_abnormal": return String(localized: "Find any abnormal values across my health records and tell me which ones are worth asking about.")
        case "compare_reports": return String(localized: "Compare my latest report with the previous one. What changed, and does the direction matter?")
        case "analyte_trend": return String(localized: "Chart how my key lab values have changed over time and explain the trend in plain language.")
        case "questions_for_doctor": return String(localized: "Based on my records, what should I ask at my next appointment? Keep it to a short list I can read out.")
        case "adherence_check": return String(localized: "Am I taking my medicines on time? Show my adherence for the last month and when I most often miss a dose.")
        case "medication_schedule": return String(localized: "Walk me through my current medicines and when I am meant to take each one.")
        case "missed_dose_pattern": return String(localized: "Which doses do I miss most often, and what would make them easier to remember?")
        case "reach_my_goal": return String(localized: "At my current rate, when do I reach my goal weight? Chart the projection and tell me what would move it.")
        case "month_over_month": return String(localized: "Compare this month with last month across everything you can see, and chart what changed most.")
        case "one_thing_to_change": return String(localized: "If I could only change one habit this month, what does my data say it should be?")
        case "whole_picture": return String(localized: "Put my whole picture together: intake, movement, sleep, labs and medicines. What is the headline?")
        default: return nil
        }
    }

    static func category(_ name: String) -> String? {
        switch name {
        case "nutrition": return String(localized: "Nutrition")
        case "sleep": return String(localized: "Sleep & recovery")
        case "training": return String(localized: "Training")
        case "labs": return String(localized: "Labs & reports")
        case "medications": return String(localized: "Medications")
        case "planning": return String(localized: "Trends & planning")
        default: return nil
        }
    }
}

File: android/app/src/main/java/com/ayuvo/health/ui/coach/PromptGalleryText.kt
package com.ayuvo.health.ui.coach

import androidx.annotation.StringRes
import com.ayuvo.health.R

/**
 * The gallery's text, keyed by catalog id (docs/coach.md §9).
 *
 * The ids and the English source live in `shared/coach/prompt_gallery.json`; this table only
 * points each one at its string resource, so a card, a chip and a translation can never drift
 * apart. `PromptGalleryTextTest` compares the resources with the catalog.
 */
object PromptGalleryText {
    @StringRes
    fun title(id: String): Int? = when (id) {
        "diet_chart_from_health" -> R.string.coach_prompt_diet_chart_from_health_title
        "macro_balance" -> R.string.coach_prompt_macro_balance_title
        "dinner_tonight" -> R.string.coach_prompt_dinner_tonight_title
        "protein_sources" -> R.string.coach_prompt_protein_sources_title
        "week_review" -> R.string.coach_prompt_week_review_title
        "improve_sleep_cycle" -> R.string.coach_prompt_improve_sleep_cycle_title
        "sleep_week" -> R.string.coach_prompt_sleep_week_title
        "sleep_consistency" -> R.string.coach_prompt_sleep_consistency_title
        "sleep_and_resting_hr" -> R.string.coach_prompt_sleep_and_resting_hr_title
        "training_review" -> R.string.coach_prompt_training_review_title
        "moving_enough" -> R.string.coach_prompt_moving_enough_title
        "lift_progress" -> R.string.coach_prompt_lift_progress_title
        "recovery_check" -> R.string.coach_prompt_recovery_check_title
        "explain_latest_report" -> R.string.coach_prompt_explain_latest_report_title
        "find_abnormal" -> R.string.coach_prompt_find_abnormal_title
        "compare_reports" -> R.string.coach_prompt_compare_reports_title
        "analyte_trend" -> R.string.coach_prompt_analyte_trend_title
        "questions_for_doctor" -> R.string.coach_prompt_questions_for_doctor_title
        "adherence_check" -> R.string.coach_prompt_adherence_check_title
        "medication_schedule" -> R.string.coach_prompt_medication_schedule_title
        "missed_dose_pattern" -> R.string.coach_prompt_missed_dose_pattern_title
        "reach_my_goal" -> R.string.coach_prompt_reach_my_goal_title
        "month_over_month" -> R.string.coach_prompt_month_over_month_title
        "one_thing_to_change" -> R.string.coach_prompt_one_thing_to_change_title
        "whole_picture" -> R.string.coach_prompt_whole_picture_title
        else -> null
    }

    @StringRes
    fun prompt(id: String): Int? = when (id) {
        "diet_chart_from_health" -> R.string.coach_prompt_diet_chart_from_health_text
        "macro_balance" -> R.string.coach_prompt_macro_balance_text
        "dinner_tonight" -> R.string.coach_prompt_dinner_tonight_text
        "protein_sources" -> R.string.coach_prompt_protein_sources_text
        "week_review" -> R.string.coach_prompt_week_review_text
        "improve_sleep_cycle" -> R.string.coach_prompt_improve_sleep_cycle_text
        "sleep_week" -> R.string.coach_prompt_sleep_week_text
        "sleep_consistency" -> R.string.coach_prompt_sleep_consistency_text
        "sleep_and_resting_hr" -> R.string.coach_prompt_sleep_and_resting_hr_text
        "training_review" -> R.string.coach_prompt_training_review_text
        "moving_enough" -> R.string.coach_prompt_moving_enough_text
        "lift_progress" -> R.string.coach_prompt_lift_progress_text
        "recovery_check" -> R.string.coach_prompt_recovery_check_text
        "explain_latest_report" -> R.string.coach_prompt_explain_latest_report_text
        "find_abnormal" -> R.string.coach_prompt_find_abnormal_text
        "compare_reports" -> R.string.coach_prompt_compare_reports_text
        "analyte_trend" -> R.string.coach_prompt_analyte_trend_text
        "questions_for_doctor" -> R.string.coach_prompt_questions_for_doctor_text
        "adherence_check" -> R.string.coach_prompt_adherence_check_text
        "medication_schedule" -> R.string.coach_prompt_medication_schedule_text
        "missed_dose_pattern" -> R.string.coach_prompt_missed_dose_pattern_text
        "reach_my_goal" -> R.string.coach_prompt_reach_my_goal_text
        "month_over_month" -> R.string.coach_prompt_month_over_month_text
        "one_thing_to_change" -> R.string.coach_prompt_one_thing_to_change_text
        "whole_picture" -> R.string.coach_prompt_whole_picture_text
        else -> null
    }

    @StringRes
    fun category(name: String): Int? = when (name) {
        "nutrition" -> R.string.coach_prompt_category_nutrition
        "sleep" -> R.string.coach_prompt_category_sleep
        "training" -> R.string.coach_prompt_category_training
        "labs" -> R.string.coach_prompt_category_labs
        "medications" -> R.string.coach_prompt_category_medications
        "planning" -> R.string.coach_prompt_category_planning
        else -> null
    }
}
